### PR TITLE
Clean-up unused imports and standardized order

### PR DESCRIPTION
**Pull request by zm711**
Inspired by @h-mayorquin and his cleaning up of imports on spikeinterface.

Also we use scipy for 3 functions total and the top import is really slowing down the import time. I moved the import to those functions since I don't know how often people actually use them.

### CODE REVIEW

**Comment by zm711** [comment]
For additional reference we can see that a huge chunk of import time is eaten by importing scipy.signal. It seems about 80ish or so percent of importing neo (through spikeinterface) is coming from the scipy import.

image

File: neo/core/analogsignal.py [hunk 1]
```diff
@@ -612,8 +604,9 @@ def downsample(self, downsampling_factor, **kwargs):
         -----
         For resampling the signal with a fixed number of samples, see `resample` method.
         """
-
-        if not HAVE_SCIPY:
+        try:
+            import scipy.signal
```
Comment (zm711) [inline neo/core/analogsignal.py:608]: Import only here so cost if sonly when people want to use the function. Which I assume is rarer than importing the package overall.

File: neo/io/__init__.py [hunk 2]
```diff
@@ -281,7 +281,7 @@
 # if it is not present, use the neurosharectypesio to load files
 try:
     import neuroshare as ns
-except ImportError as err:
+except ModuleNotFoundError as err:
```
Comment (zm711) [inline neo/io/__init__.py:284]: Seems like this is the more modern error for this? Happy to revert though...

File: neo/io/tools.py [hunk 3]
```diff
@@ -4,12 +4,8 @@
     SPikeTrains
 """
 
-try:
-    from collections.abc import MutableSequence
-except ImportError:
-    from collections import MutableSequence
 
-import numpy as np
+from collections.abc import MutableSequence
```
Comment (zm711) [inline neo/io/tools.py:8]: As of 3.7 this is the only way to do this, so since 3.8 is our min this change is fine.

**Review by zm711 — COMMENTED** [review]
Notes for people to check as we review.

**Review by samuelgarcia — APPROVED** [review]

**Comment by samuelgarcia** [comment]
Merci Zach

**Pull request merged**